cmd/gardenlet/app/app.go

```diff
@@ -460,6 +468,38 @@ func (g *garden) cleanupOrphanedExtensionsServiceAccounts(ctx context.Context, g
 	return flow.Parallel(taskFns...)(ctx)
 }
 
+// TODO(Kostov6): Remove this code after v1.91 has been released.
+func (g *garden) cleanupGRMSecretFinalizers(ctx context.Context, seedClient client.Client) error {
```

Comment:
Instead of `g.mgr.GetClient()` you can use the `seedClient`.

```suggestion
func cleanupGRMSecretFinalizers(ctx context.Context, seedClient client.Client) error {
```